### DISCUSSION TITLE
What is the correct syntax when using the client.query function?

### DISCUSSION THREAD

**Discussion by jsoneaday** [urql-graphql/urql#1788] [Q&A] [answered]
Hello I am using the useQuery React hook without issue, but when I switch to using the core call client.query I get the error GraphQLError: Syntax Error: Expected Name, found "$"

```
const query = createRequest(
      gql`{
      query Transactions($tags: [TagFilter!]) {
        transactions(tags: $tags) {
          edges {
            node {
              id
              tags {
                name
                value
              }
            }
          }
        }
      }
    }`,
      {
        tags: [
          { name: "App-Name", values: ["DeChat"] },
          { name: "Type", values: [DeChatTypes.message] },
          { name: "walletAddress", values: [userPublicWalletAddress] },
        ],
      }
    );
    const messagesGql = await client.query(query.query).toPromise();
```

**Comment by JoviDeCroock** [comment]
`client.query(yourQuery, yourVariables, yourContext)` as documented [here](https://formidable.com/open-source/urql/docs/basics/core/#one-off-queries-and-mutations)

**Reply by jsoneaday** [reply]
The two last variables are optional. Also the createRequest function takes the variables as its second parameter, which I am passing. TypeScript provides this signature, which clearly shows variables and context are optional.

Client.query<any, object>(query: string | DocumentNode | TypedDocumentNode<any, object>, variables?: object | undefined, context?: Partial<OperationContext> | undefined): PromisifiedSource<...>

**Reply by JoviDeCroock** [reply]
yes so the types tell you that it's a `query` (docNode | string) and not a `Request`. [executeQuery does accept that](https://github.com/FormidableLabs/urql/blob/main/packages/core/src/client.ts#L118)

**Reply by jsoneaday** [reply]
Please look again as that's what I am passing. Notice I pass _query.query_. The _query_ property is a DocumentNode | TypedDocumentNode<any, object>. This is why my failure is not happening during development. Because if I was passing the wrong parameter type during development obviously the compiler would not allow compilation since I am using TypeScript. The error given happens during runtime.

**Comment by kitten** [comment, marked as answer]
I think you may have simply typo'd the query; the error says _GraphQLError_ and hence seems like an API / parsing error, so typically please be precise with where the error is coming from and trace it first 😅 

You start the query with

```
{
      query Transactions($tags: [TagFilter!]) {
```

Which is the shortcut syntax mixed with the query definition, so the first opening `{` curly brace is the issue
